mlos_core/mlos_core/optimizers/bayesian_optimizers.py

```diff
@@ -188,9 +190,9 @@ def __init__(
         elif base_estimator == 'gp':
             self._transform = self._to_numeric
         else:
-            self._transform = np.array
+            self._transform = self._to_ndarray
```

Comment:
can we just use `pandas.DataFrame.to_numpy` here instead? Can you please try? If it works, we won't need to define `_to_numpy()` at all

Reply:
I tried this, but the signature for `DataFrame.to_numpy` actually accepts a whole pile of other types as well and mypy gets grumpy about it.
For now, I simplified the function I was using (and moved it out of the class to make pylint happy) to basically just call `.to_numpy` on the config object.  Not sure why this works, and a lambda version of the same doesn't, but *shrug*.
I think I'll commit this as is for now and we can play with it more later.  Cool?